cloudsync_onedrive.py

```diff
@@ -446,7 +465,7 @@ def _fetch_shared_drives(self):
 
     def _fetch_sites(self):
         # sharepoint sites - a user can have access to multiple sites, with multiple drives in each
-        sites = self._direct_api("get", "/sites?search=*").get("value", [])
+        sites = self._safe_direct_api("/sites?search=*", default={}).get("value", [])
```

Comment:
`_fetch_sites` would have previously raised an exception to the client, which maybe is not ideal, but it would have also prevented the flag which finalizes the various caches from getting set. Now, we will have no cached sites, and no way to detect the problem, and no direct way to repopulate the cache.

Reply:
`list_ns` and `_get_validated_namespace` are both impacted by the potentially empty cache of sites, producing new behaviors that I'm not sure are what we want

Reply:
failing the operation and having the caller try-again seems much safer than returning partial info

Reply:
Certainly, but we've seen errors/scenarios where "try again" would not help, in those cases our only option is to operate in a degraded state (where some namespaces are missing) -- otherwise there is no way to complete the connection process.

Some specific examples: 
- 1 of 45 sites a user has access to returns with a missing attribute (according to MS docs, the attribute is not optional)
- sites search API call returns a 404

We can collect/expose the suppressed errors to be more explicit.

Reply:
Update: now collecting namespace errors in a special namespace that can be queried via `list_ns` (see test)